src/MonC/Semantics/SemanticAnalyzer.cs

```diff
@@ -40,21 +46,10 @@ public void Analyze(ParseModule headerModule, ParseModule newModule)
             foreach (EnumNode enumNode in headerModule.Enums) {
                 RegisterEnum(enumNode);
             }
-            foreach (EnumNode enumNode in newModule.Enums) {
-                RegisterEnum(enumNode);
-            }
 
             foreach (FunctionDefinitionNode externalFunction in headerModule.Functions) {
                 _functions.Add(externalFunction.Name, externalFunction);
             }
-            foreach (FunctionDefinitionNode function in newModule.Functions) {
-                if (_functions.TryGetValue(function.Name, out FunctionDefinitionNode existingFunction)) {
-                    if (!ReferenceEquals(function, existingFunction)) {
-                        _errorsToProcess.Add(("Redefinition of function " + function.Name, function));
```

Comment:
I debated whether or not to keep this in. Lots of redundant checking is avoided by removing it and relying on the `HeaderModuleBuilder` to do its job.